Source/Core/Common/CodeBlock.h

```diff
@@ -60,42 +63,55 @@ class CodeBlock : public T, NonCopyable
   // Call this when shutting down. Don't rely on the destructor, even though it'll do the job.
   void FreeCodeSpace()
   {
-    Common::FreeMemoryPages(region, region_size);
+    _assert_(!m_is_child);
+    Common::FreeMemoryPages(region, total_region_size);
     region = nullptr;
     region_size = 0;
-    parent_region_size = 0;
-    if (m_has_child)
+    total_region_size = 0;
+    for (CodeBlock* child : m_children)
     {
-      m_child->region = nullptr;
-      m_child->region_size = 0;
+      child->region = nullptr;
+      child->region_size = 0;
+      child->total_region_size = 0;
     }
   }
 
-  bool IsInSpace(const u8* ptr) const { return ptr >= region && ptr < (region + region_size); }
+  bool IsInSpace(const u8* ptr) const
+  {
+    return ptr >= region && ptr < (region + total_region_size);
```

Comment:
I changed the meaning of `region_size`. It originally meant what `total_region_size` means now.

I did that because `parent_region_size` was a confusing name.